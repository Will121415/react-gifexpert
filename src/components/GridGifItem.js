import React from 'react'
import { PropTypes } from 'prop-types'
export const GridGifItem = ({url,title}) => {
  return (
    <div className='card'>
        <img src={url} alt={title} />
        <p>{title}</p>
    </div>
  )
}

GridGifItem.propTypes = {
  url: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired
}