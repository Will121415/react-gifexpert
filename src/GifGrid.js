import React from 'react'
import { GridGifItem } from './GridGifItem';
import { useFetchGifs } from './Hooks/useFetchGifs';

export const GifGrid = ({ category }) => {

    const {data:images, loading} = useFetchGifs(category);

    console.log(category, images, loading);


    return (
        <>
            <h3>{category}</h3>
            {loading && <strong>Loading...</strong>}
            <div className='card-grid animate__animated animate__fadeIn'>
                {
                    images.map(img => (
                        <GridGifItem
                            key={img.id}
                            {...img}
                        />
                    ))
                }
            </div>
        </>

    )
}
