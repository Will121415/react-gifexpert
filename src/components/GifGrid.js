import React from 'react'
import { GridGifItem } from './GridGifItem';
import { useFetchGifs } from '../Hooks/useFetchGifs';
import { PropTypes } from 'prop-types';

export const GifGrid = ({ category }) => {

    const {data:images, loading} = useFetchGifs(category);

    // console.log(category, images, loading);


    return (
        <>
            <h3>{category}</h3>
            {loading && <strong>Loading...</strong>}
            <div className='card-grid animate__animated animate__fadeIn'>
                {
                    images.map(img => (
                        <GridGifItem
                            key={img.id}
                            {...img}
                        />
                    ))
                }
            </div>
        </>

    )
}

GifGrid.propTypes = {
    category: PropTypes.string.isRequired
}