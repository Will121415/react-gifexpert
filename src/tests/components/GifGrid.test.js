import { shallow } from 'enzyme';
import { GifGrid } from '../../components/GifGrid';
import { useFetchGifs } from '../../Hooks/useFetchGifs';
import '@testing-library/jest-dom';
jest.mock('../../Hooks/useFetchGifs');

describe('Pruebas en el componente <GifGrid />', () => {

    const category = 'One Punch';

    test('Debe funcionar correctamente', () => { 
        useFetchGifs.mockReturnValue({
            data: [],
            loading: true
        })
        const wapper = shallow(<GifGrid category={category}/>);
        expect(wapper).toMatchSnapshot();
    })

    test('Debe mostrar items cuando se cargan las imagenes useFetchGifs', () => {

        const gifs = [{
            id: 'ABC',
            url: 'https://localhost/cualquiercosa.jpg',
            title: 'Cualquier cosa'
        }]

        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false
        })

        const wapper = shallow(<GifGrid category={category}/>);

        // expect(wapper).toMatchSnapshot();
        expect(wapper.find('strong').exists()).toBe(false);
        expect(wapper.find('GridGifItem').length).toBe(gifs.length); 

    })
})