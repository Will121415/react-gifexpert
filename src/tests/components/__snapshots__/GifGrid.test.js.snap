// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente <GifGrid /> Debe funcionar correctamente 1`] = `
<Fragment>
  <h3>
    One Punch
  </h3>
  <strong>
    Loading...
  </strong>
  <div
    className="card-grid animate__animated animate__fadeIn"
  />
</Fragment>
`;
